### ISSUE TITLE
Accessibility Hook

### ISSUE THREAD

**Issue by korgon** [searchspring/snap#708]
Keyboard navigation and accessibility functionality should be easily added inside any component.

The hook should allow for tab navigation on specific elements as well as applying any other best practices to make our components more accessible - this may include things like aria-attributes or similar. Spatial navigation is also something that we  would like to support to allow navigating within a component via arrow keys.

Starting with 'dropdown' / 'select' components.
Eventually moving to 'carousel' / 'recommendation' components and 'facets' / 'facet' as well as 'autocomplete' and any other components that may need accessibility.
